resources/todomvc/big-dom-generator/src/components/tree-area.jsx

```diff
@@ -1,30 +1,24 @@
-import { LCG } from "random-seedable";
-import { useRef } from "react";
-import { DEFAULT_SEED_FOR_RANDOM_NUMBER_GENERATOR, MAX_GENERATED_DOM_DEPTH, MAX_NUMBER_OF_CHILDREN, PROBABILITY_OF_HAVING_CHILDREN, TARGET_SIZE, MAX_VISIBLE_TREE_VIEW_ITEM_DEPTH } from "../../params";
+import { MAX_VISIBLE_TREE_VIEW_ITEM_DEPTH } from "../../params";
+import { generateTreeHead } from "./../tree-generator";
 
 import ChevronRight from "./../assets/Smock_ChevronRight_18_N.svg";
 import TaskListIcon from "./../assets/Smock_TaskList_18_N.svg";
 
 const FolderWrapper = (props) => {
-    const { nodeCount, random, maxDepth, maxNumChildren, childProbability, currentDepth } = props;
-    // prettier-ignore
-    if (currentDepth >= maxDepth)
-        return null;
-    // Choose a random number of children.
-    const numChildren = random.randRange(1, maxNumChildren);
+    const { treeNode, currentDepth } = props;
+
     const children = [];
-    for (let i = 0; i < numChildren && nodeCount.current < TARGET_SIZE; i++)
-        children.push(<TreeItem key={i} nodeCount={nodeCount} random={random} numChildren={numChildren} maxNumChildren={maxNumChildren} maxDepth={maxDepth} childProbability={childProbability} currentDepth={currentDepth + 1} />);
+    treeNode.children.forEach((child) => {
+        children.push(<TreeItem treeNode={child} currentDepth={currentDepth + 1} />);
+    });
 
-    nodeCount.current = nodeCount.current + 1;
     return <ul className="spectrum-TreeView spectrum-TreeView--sizeS">{children}</ul>;
 };
 
 const TreeItem = (props) => {
-    const { nodeCount, random, maxDepth, maxNumChildren, childProbability, currentDepth } = props;
-    nodeCount.current = nodeCount.current + 4;
-    // Choose whether to have children.
-    const children = random.coin(childProbability) ? <FolderWrapper nodeCount={nodeCount} random={random} maxNumChildren={maxNumChildren} maxDepth={maxDepth} childProbability={childProbability} currentDepth={currentDepth + 1} /> : null;
+    const { treeNode, currentDepth } = props;
+
+    const children = treeNode.children.length ? <FolderWrapper treeNode={treeNode.children[0]} currentDepth={currentDepth + 1} /> : null;
```

Comment:
We always push the first child of treeNode because we know that treeNode.children.length is 1 if there are children. I'm adding a comment for this in the code.
```js
currentNode.children.push({ type: type, children: [] });
currentNode = currentNode.children[currentNode.children.length - 1];
```
I wasn't able to repro the behavior of only seeing 7 as the max depth, which complex example did you run?

Reply:
I think I looked at the React example.

Reply:
> We always push the first child of treeNode because we know that treeNode.children.length is 1 if there are children. 

It's not clear to me why we have this array of 1 item then... This is very confusing to me... But I'll look at the updated code that might clear things up.

Reply:
Does an expandableItem always contains 1 item that is always a list? I feel like we could combine them into one then...

We could have simply:
* list: this is expandable
* item: this is non-expandable

Another option is to only have nodes with a children array, no `type`:
* children is empty => non-expandable item
* children is non empty => expandable item

These 2 structures are common ways to represent a tree. I'm not sure why we have something different here.

Reply:
I read the code again and I understand that you want the 3 types: expanded, not expanded, normal item. I believe part of my confusion comes from the word "expandable". Indeed the term you want is rather "expanded".

Here is another structure proposal:
* normal: `{ type: 'item' }`
* expandedItem: `{ type: 'expandedItem', children: [ item1, item2, ... ] }` // Note: no array with just 1 list item!
* nonExpandedItem: `{ type: 'nonExpandedItem' }` // In a real app, this on would also have children because the user would be able to expand it

But please also make sure that `expandedItem` have at least 1 child !!

Reply:
@julienw we did mean expandable (representing a Sprint Folder) and non-expandable (representing a Todo List), but I understand where your confusion is coming from, we've been dragging some unnecessary complexity all the way from the initial version of this code. This is how we'll simplify it:

On the react file:

1. We will remove the folder wrapper component.
2. We will modify the `TreeItem` component. Each `TreeItem` component will be either expandable or non-expandable.
3. An expandable `TreeItem` will be of the form:
```
<li>
  <a>
    <svg/>
    <span/>
  <a>
  <ul>
    {children>
  <ul/>
<li>
``` 
Note that an expandable item can have 0 children (empty folder).
4. A non-expandable `TreeItem` will be of the form:
```
<li>
  <a>
    <svg/>
    <span/>
  <a>
<li>
```

The generator will change so that:

1. We only have 2 types of nodes: `expandableItem` and `nonExpandableItem`. Note that `nonExpandableItem` can't have children, but `expandableItem` may or may not have them.
2. `expandableItemWeight` will account for the `ul` node.

This will simplify the logic of which nodes should of not should have children.